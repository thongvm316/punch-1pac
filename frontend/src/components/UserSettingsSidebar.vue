<template>
  <ul class="setting-sidebar">
    <li
      :class="{ active: activeTabName === 'Profile' }"
      @click="setActive('Profile')"
    >
      <p-ico-user />
      {{ $t('sidebar.user.profile') }}
    </li>
    <li
      :class="{ active: activeTabName === 'Password' }"
      @click="setActive('Password')"
    >
      <p-ico-key />
      {{ $t('sidebar.user.password') }}
    </li>
    <li
      :class="{ active: activeTabName === 'Security' }"
      @click="setActive('Security')"
    >
      <p-ico-shield />
      {{ $t('sidebar.user.security') }}
    </li>
  </ul>
</template>

<script>
import activeTab from '@/mixins/active-tab'
const PIcoUser = () => import('@/punch-ui/punch-icons/PIcoUser')
const PIcoKey = () => import('@/punch-ui/punch-icons/PIcoKey')
const PIcoShield = () => import('@/punch-ui/punch-icons/PIcoShield')

export default {
  name: 'UserSettingsSidebar',

  components: {
    PIcoUser,
    PIcoKey,
    PIcoShield
  },

  mixins: [activeTab]
}
</script>
