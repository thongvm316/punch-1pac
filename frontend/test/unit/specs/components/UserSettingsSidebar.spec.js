import { shallowMount } from '@vue/test-utils'
import wrapperOps from '../../supports/wrapper'
import UserSettingsSidebar from '@/components/UserSettingsSidebar'

const localWrapperOps = {
  ...wrapperOps,
  stubs: {
    PIcoUser: true,
    PIcoKey: true,
    PIcoShield: true
  }
}
const setActive = jest.spyOn(UserSettingsSidebar.mixins[0].methods, 'setActive')

describe('UserSettingsSidebar.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(UserSettingsSidebar, localWrapperOps)
  })

  afterEach(() => { wrapper.vm.$destroy() })

  describe('when UserSettingsSidebar mounted', () => {
    it('should render correctly', () => {
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper.isVueInstance()).toBeTruthy()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when setActive', () => {
    it('should active Holidays tab', async () => {
      wrapper.findAll('li').at(2).trigger('click')
      await wrapper.vm.$nextTick()

      expect(setActive).toHaveBeenCalledWith('Security')
      expect(wrapper).toMatchSnapshot()
    })

    it('should emit @active', async () => {
      wrapper.findAll('li').at(2).trigger('click')
      await wrapper.vm.$nextTick()

      expect(wrapper.emitted().active).toHaveLength(1)
      expect(wrapper.emitted().active[0]).toEqual(['Security'])
    })
  })
})
