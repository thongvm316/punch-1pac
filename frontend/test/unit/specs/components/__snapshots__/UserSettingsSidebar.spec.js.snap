// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSettingsSidebar.vue when UserSettingsSidebar mounted should render correctly 1`] = `
<ul class="setting-sidebar">
  <li class="active">
    <picouser-stub></picouser-stub>
    Profile
  </li>
  <li class="">
    <picokey-stub></picokey-stub>
    Password
  </li>
  <li class="">
    <picoshield-stub></picoshield-stub>
    Security
  </li>
</ul>
`;

exports[`UserSettingsSidebar.vue when setActive should active Holidays tab 1`] = `
<ul class="setting-sidebar">
  <li class="">
    <picouser-stub></picouser-stub>
    Profile
  </li>
  <li class="">
    <picokey-stub></picokey-stub>
    Password
  </li>
  <li class="active">
    <picoshield-stub></picoshield-stub>
    Security
  </li>
</ul>
`;
